examples/unimp/unimp_trainer.py

```diff
@@ -0,0 +1,168 @@
+import os
+os.environ['CUDA_VISIBLE_DEVICES']='0'
+import random
+import argparse
+import tensorlayerx as tlx
+import tensorlayerx.nn as nn
+from gammagl.utils import segment_softmax
+from gammagl.datasets import Planetoid
+from gammagl.layers.conv import MessagePassing
+from gammagl.utils import add_self_loops, mask_to_index
+from tensorlayerx.model import TrainOneStep, WithLoss
+
+class CrossEntropyLoss(WithLoss):
+    def __init__(self, model, loss_func):
+        super(CrossEntropyLoss, self).__init__(model,loss_func)
+
+    def forward(self, data, label):
+        out = self.backbone_network(data['x'], data['edge_index'])
+        out = tlx.gather(out, data['val_idx'])
+        label = tlx.reshape(tlx.gather(label, data['val_idx']),shape=(-1,))
+        #print(out[0])
+        #print(label[0])
+        loss = self._loss_fn(out, label)
+        return loss
+
+
+class MultiHead(MessagePassing):
+    def __init__(self, in_features, out_features, n_heads,num_nodes):
+        super().__init__()
+        self.heads=n_heads
+        self.num_nodes=num_nodes
+        self.out_channels=out_features
+        self.linear = tlx.layers.Linear(out_features=out_features* n_heads,
+                                        in_features=in_features)
+
+        init = tlx.initializers.RandomNormal()
+        self.att_src = init(shape=(1, n_heads, out_features), dtype=tlx.float32)
+        self.att_dst = init(shape=(1, n_heads, out_features), dtype=tlx.float32)
+
+        self.leaky_relu = tlx.layers.LeakyReLU(0.2)
+        self.dropout = tlx.layers.Dropout()
+
+    def message(self, x, edge_index):
+        node_src = edge_index[0, :]
+        node_dst = edge_index[1, :]
+        weight_src = tlx.gather(tlx.reduce_sum(x * self.att_src, -1), node_src)
+        weight_dst = tlx.gather(tlx.reduce_sum(x * self.att_dst, -1), node_dst)
+        weight = self.leaky_relu(weight_src + weight_dst)
+
+        alpha = self.dropout(segment_softmax(weight, node_dst, self.num_nodes))
+        x = tlx.gather(x, node_src) * tlx.expand_dims(alpha, -1)
+        return x
+
+
+    def forward(self, x, edge_index):
+        x = tlx.reshape(self.linear(x), shape=(-1,self.heads, self.out_channels))
+        x = self.propagate(x, edge_index, num_nodes=self.num_nodes)
+        x=tlx.ops.reduce_mean(x,axis=1)
+
+        return x
+
+
+class Unimp(tlx.nn.Module):
```

Comment:
Put this section into ``gammagl.models.unimp.py``. If you put this into this file, users will not be able to use this function.